bindsnet/network/nodes.py

```diff
@@ -1218,7 +1218,7 @@ def forward(self, x: torch.Tensor) -> None:
 
         # Add inter-columnar input.
         if self.s.any():
-            x += self.S[:, self.s].sum(dim=1)
+            x += self.S[:, self.s.view(-1)].sum(dim=1)
```

Comment:
I don't think this will work if you use minibatching. Spikes will have shape `[batch_size, ...]`; I'm guessing you're using `batch_size = 1` in you use case. To get it to work with arbitrary minibatch size, you should replace it with:

```
torch.cat([S[:, s[i]].sum(dim=1)[None] for i in range(s.shape[0])], dim=0)
```

Or a vectorized equivalent of that.